include/gridtools/fn/backend2/gpu.hpp

```diff
@@ -33,8 +33,9 @@ namespace gridtools::fn::backend {
 
         template <class BlockSizes, class Sizes>
         GT_FUNCTION_DEVICE auto global_thread_index() {
-            using ndims_t = meta::length<Sizes>;
-            using keys_t = meta::take_c<std::min(3, (int)ndims_t::value), get_keys<Sizes>>;
+            using all_keys_t = get_keys<Sizes>;
```

Comment:
(In line 28) Can you add documentation of what the type of `BlockSizes` should be?

Reply:
Done.